feat: Added the my feed position variations and stylings for each version

rebelchris opened the pull request:
## Changes

- Added the AB test for the my feed position
- Refactored how we pass the feed filters on the feed pages
- Added specific styling for each type

![image](https://user-images.githubusercontent.com/554874/152804249-c4fc5236-3008-441c-a658-6ab207ed2b7f.png)

## Manual Testing

- On those affected packages:
- [x] Have you done sanity checks in the webapp?
- [x] Have you done sanity checks in the extension?

DD-529 #done

idoshamun commented on packages/shared/src/hooks/useMyFeed.ts:
```diff
@@ -25,6 +30,7 @@ export function useMyFeed(): UseMyFeed {
   const { user } = useContext(AuthContext);
   const { flags } = useContext(FeaturesContext);
   const shouldShowMyFeed = isFeaturedEnabled(Features.MyFeedOn, flags);
+  const myFeedPosition = getFeatureValue(Features.MyFeedPosition, flags);
```
We want to use our new function that chooses a value out of a set of possibilities otherwise fall back to the default `sidebar`. We might introduce new variations in the future and don't want the version to break.

rebelchris replied:
This is the case, the defaults are managed in the `featureManagement` class.

See here: 

https://github.com/dailydotdev/apps/blob/1dee521b54563cb0e9224dc2a9fb7a2cad1561d2/packages/shared/src/lib/featureManagement.ts#L33

```js
  static readonly MyFeedPosition = new Features('my_feed_position', 'sidebar');
```

idoshamun replied:
It's not enough actually. Let's say in this version we support two values for this feature `sidebar` and `feed_ad` (I know that in reality there's more but just an example). Then the next version comes and we want to add a new type `header`. We set it in Flagsmith and now the old version breaks. Whereas if we check in every version of the received value is one of the supported values and if not fall back to default, it will be backward compatible.

idoshamun reviewed (changes requested):
Sweet! A few changes before we merge

rebelchris commented:
@idoshamun Thanks for that amazing review!
The possibility fallback should already be in place, the other points I've addressed 🙌

sshanzel commented on packages/shared/src/components/CreateMyFeedButton.tsx:
```diff
@@ -0,0 +1,129 @@
+    feed_ad: `${explainerColor.shadow} p-2 border flex-col flex-1 justify-center`,
+  };
+};
+const textClass = (sidebarExpanded: ClassProps) => {
```
I think this needs destructuring.
```suggestion
const textClass = ({ sidebarExpanded }: ClassProps) => {
```

sshanzel commented on packages/shared/src/components/CreateMyFeedButton.tsx:
```diff
@@ -0,0 +1,129 @@
+  explainerColor?: ThemeColor;
+  sidebarExpanded?: boolean;
+}
+const wrapperClasses = (sidebarExpanded: ClassProps) => {
```
I think this needs destructuring.
```suggestion
const wrapperClasses = ({ sidebarExpanded }: ClassProps) => {
```

sshanzel commented on packages/shared/src/components/CreateMyFeedButton.tsx:
```diff
@@ -0,0 +1,129 @@
+}
+const wrapperClasses = (sidebarExpanded: ClassProps) => {
+  return {
+    sidebar: sidebarExpanded ? 'h-[8.125rem] justify-center' : 'h-[8.125rem] ',
```
Both of them have the same height, I think we can do the condition on the centering part only.
```suggestion
    sidebar: classNames('h-[8.125rem]', sidebarExpanded && 'justify-center'),
```

idoshamun commented on packages/shared/src/components/CreateMyFeedButton.tsx:
```diff
@@ -0,0 +1,128 @@
+
+const getAnalyticsEvent = (
+  eventName: string,
+  copy: string,
```
Yes :)

sshanzel commented on packages/shared/src/components/CreateMyFeedButton.tsx:
```diff
@@ -0,0 +1,129 @@
+};
+const textClass = (sidebarExpanded: ClassProps) => {
+  return {
+    sidebar: sidebarExpanded
+      ? 'typo-footnote w-[11.25rem] mb-3 transform opacity-100 ease-linear duration-200  delay-200'
+      : 'typo-footnote w-[11.25rem] mb-3 transform duration-0 delay-0 opacity-0',
```
I think we can unify some of the classes.
```suggestion
    sidebar: classNames('typo-footnote w-[11.25rem] mb-3 transform', sidebarExpanded
      ? 'opacity-100 ease-linear duration-200  delay-200'
      : 'duration-0 delay-0 opacity-0'),
```

sshanzel reviewed (changes requested):
So awesome to see that we have a reusable component for different scenarios! Just had a few comments but nothing major 🚢

rebelchris commented:
@sshanzel and @idoshamun thanks so much!
Added a validTypes object that can be set optional.

This should also help with defining what types we allow for each of the features.
And added some basic tests to test if nothing breaks over time in this feature.

sshanzel reviewed (approved):
LGTM! 🚢

idoshamun reviewed (approved):
Awesome!
